pkg/plugin/plugin.go

```diff
@@ -22,6 +25,65 @@ type plugin struct {
 
 var _ api.Plugin = &plugin{}
 
+func getEnv(name string, defaultValue ...string) string {
+	value := os.Getenv(name)
+	if len(value) == 0 {
+		if len(defaultValue) > 0 {
+			return defaultValue[0]
+		}
+		return ""
+	}
+	return value
+}
+
+// GetRequiredConfigEnvVars exposes the required variables to build a plugin config
+func GetRequiredConfigEnvVars() []string {
+	return []string{
+		"AZURE_TENANT_ID",
+		"AZURE_SUBSCRIPTION_ID",
+		"AZURE_CLIENT_ID",
+		"AZURE_CLIENT_SECRET",
+		"RESOURCEGROUP",
+	}
+}
+
+// NewPluginConfigFromEnv loads all config items from os env vars
+func NewPluginConfigFromEnv() (api.PluginConfig, error) {
+	//log.Info("building plugin config from environment variables")
+
+	pc := api.PluginConfig{}
+	required := GetRequiredConfigEnvVars()
+
+	var missing []string
+	for i := range required {
+		value := os.Getenv(required[i])
+		if len(value) == 0 {
+			missing = append(missing, required[i])
+		}
+	}
+
+	if len(missing) > 0 {
```

Comment:
There's a pretty cool struct validator package in https://github.com/go-playground/validator maybe we can use that instead?

Reply:
Unfortunately vendoring that project will bring in the whole world because they include translations in their repo. In general, we should not be importing libraries not used by the rest of the openshift/kubernetes ecosystem without a very good reason.

Reply:
ok good to know thanks @kargakis was about to ask in channel :+1: